src/Node/Internal.hs

```diff
@@ -239,14 +239,18 @@ closeChannel (ChannelOut conn) = NT.close conn
 
 -- | Do multiple sends on a 'ChannelOut'.
 writeMany
-    :: ( Monad m, Mockable Throw m )
+    :: forall m .
+       ( Monad m, Mockable Throw m )
     => Word32 -- ^ Split into chunks of at most this size in bytes. 0 means no split.
     -> ChannelOut m
     -> LBS.ByteString
     -> m ()
-writeMany mtu (ChannelOut conn) bss =
-    NT.send conn (chop bss >>= LBS.toChunks) >>= either throw pure
+writeMany mtu (ChannelOut conn) bss = mapM_ sendUnit units
   where
+    sendUnit :: [BS.ByteString] -> m ()
+    sendUnit unit = NT.send conn unit >>= either throw pure
+    units :: [[BS.ByteString]]
+    units = fmap LBS.toChunks (chop bss)
```

Comment:
Yes, nice, so we'll always get frames of the max mtu size (except last) irrespective of how the lazy bytestring chunks are formed.